tests/python/test_python_frontend.py

```diff
@@ -4458,6 +4457,53 @@ def fusion_func(fd: FusionDefinition):
         for i in range(num_out):
             self.assertEqual(nvf_out[i].data_ptr(), inputs[0].data_ptr())
 
+    def test_sdpa_fwd(self):
+        def fusion_func(fd: FusionDefinition):
+            q = fd.define_tensor(
+                shape=[-1, -1, -1, -1],
+                contiguity=[True, True, True, True],
+                dtype=DataType.BFloat16,
+                is_cpu=False,
+            )
+            k = fd.define_tensor(
+                shape=[-1, -1, -1, -1],
+                contiguity=[True, True, True, True],
+                dtype=DataType.BFloat16,
+                is_cpu=False,
+            )
+            v = fd.define_tensor(
+                shape=[-1, -1, -1, -1],
+                contiguity=[True, True, True, True],
+                dtype=DataType.BFloat16,
+                is_cpu=False,
+            )
+            is_causal = fd.define_scalar(None, dtype=DataType.Bool)
+            scale = fd.define_scalar(None, dtype=DataType.Double)
+            attn, *intermediate_results = fd.ops.sdpfa_fwd(
+                q, k, v, is_causal=is_causal, scale=scale
+            )
+            fd.add_output(attn)
+
+        N, H, L, S, E = 4, 8, 16, 16, 8
+        is_causal_vals = [False, True]
+        scale_vals = [1 / E**0.5, 1e-3]
+        # Dropout value is not set (default = 0.0) to validate against torch
+        for is_causal, scale in itertools.product(is_causal_vals, scale_vals):
+            with self.subTest(is_causal=is_causal, scale=scale):
+                qkv = [
+                    torch.randn((N, H, L, E), dtype=torch.bfloat16, device="cuda:0"),
+                    torch.randn((N, H, S, E), dtype=torch.bfloat16, device="cuda:0"),
+                    torch.randn((N, H, S, E), dtype=torch.bfloat16, device="cuda:0"),
+                ]
+                # Reset the FusionCache or the fusion would not recompile for all subtests, failing checks in exec_nvfuser.
+                # TODO: Try to move this to pytest_ops.py. Currently, it does not work since the API between nvFuser and torch differs.
+                FusionCache.reset()
+                nvf_out, _ = self.exec_nvfuser(fusion_func, [*qkv, is_causal, scale])
+                ref_out = F.scaled_dot_product_attention(
+                    *qkv, is_causal=is_causal, scale=scale
```

Comment:
nitpick here: dropout default to None here. Is it because we don't reset seed so we cannot test with a real dropout value?

Just curious, we don't have to fix it in this PR. But maybe a follow up issue would be nice.

Reply:
Yes, with non-zero dropout, we cannot validate against torch.
What do you mean by resetting the seed?

Reply:
I assume that we cannot validate against torch simply because the offset is moved?

i.e. Would this allow us to validate? vvv
```
torch.cuda.manual_seed(0)
ref_out = F.scaled_dot_product_attention(...)
```

Reply:
Right, we are not validating the RNG state here so this should work.
Let me try.